src/infra/core/src/verification_service_impl.rs

```diff
@@ -119,6 +120,9 @@ impl VerificationServiceImpl {
                             },
                         ));
                     } else {
+                        if !check_logical_hashes {
+                            break;
```

Comment:
This looks like a bug. You are terminating the entire verification loop upon encountering the first block with data.